### PR TITLE
Adds SQL support in Spark Command

### PR DESCRIPTION
**Pull request by sourabh912**
Jira: https://qubole.atlassian.net/browse/SDK-8

@mindprince @PraveenSeluka

### CODE REVIEW

File: qds_sdk/commands.py [hunk 1]
```diff
@@ -295,13 +295,15 @@ def parse(cls, args):
 class SparkCommand(Command):
 
     usage = ("sparkcmd <submit|run> [options]")
-    allowedlanglist = ["python", "scala"]
+    allowedlanglist = ["python", "scala", "sql"]
```
Comment (mindprince) [inline qds_sdk/commands.py]: Is this needed?

File: qds_sdk/commands.py [hunk 2]
```diff
@@ -341,25 +343,38 @@ def validate_program(cls, options):
...
+        bool_sql = options.sql is not None
+        bool_other_options = options.script_location is not None or options.program is not None or options.cmdline is not None
+
+        # if both are false then no option is specified => raise PraseError
+        # if both are true then atleast two option specified => raise ParseError
+        if bool_sql == bool_other_options:
+            raise ParseError("Exactly One of script location or program or cmdline or sql should be specified", cls.optparser.format_help())
+        if bool_sql:
+            if options.language is None:
+                raise ParseError("Unspecified language for Sql query", cls.optparser.format_help())
```
Comment (mindprince) [inline qds_sdk/commands.py]: You don't need to send `language: sql`. So this error should be the other way around.

File: tests/test_command.py [hunk 3]
```diff
@@ -407,6 +410,32 @@ def test_submit_script_location_local_java(self):
...
+                     'program': None,
+                     'cmdline':None,
+                     'command_type': 'SparkCommand',
+                     'arguments': None,
+                     'user_program_arguments': None,
+                     'can_notify': False,
+                     'script_location': None})
+
+    def test_submit_sql_no_program(self):
+        sys.argv = ['qds.py', 'sparkcmd', 'submit', '--sql', 'show dummy']
```
Comment (mindprince) [inline tests/test_command.py]: This should be the positive test and the other one should raise an error.

File: qds_sdk/commands.py (hunk 2 above)
Comment (sourabh912) [inline qds_sdk/commands.py]: I have tried to send all the parameters which we are sending from the Analyze UI and language is one of them. I will remove 'language' and run the query again and see if it succeeds.

**Pull request closed without merging**